dracut/driver_updates.py

```diff
@@ -491,13 +492,25 @@ def _process_driver_disk(dev, interactive=False):
 
     return modules
 
-def process_driver_rpm(rpm):
+def process_driver_rpm(rpm, dev = None):
     try:
-        return _process_driver_rpm(rpm)
+        if dev:
+            return _mount_and_process_driver_rpm(rpm, dev)
+        else:
+            return _process_driver_rpm(rpm)
     except (subprocess.CalledProcessError, IOError) as e:
         log.error("ERROR: %s", e)
         return {}
 
+def _mount_and_process_driver_rpm(rpm, dev):
```

Comment:
If I read this correctly, the `mounted` context managed unmounts the device once the block is processed, right ? In such case the name of the function (`_mount_and_process_driver_rpm`) seems misleading - it sounds like the device remains mounted after the function has been called.

Sp I recommend changing changing the name to something like `_process_driver_rpm_from_device()`. :)